flowzone.yml

```diff
@@ -2139,24 +2229,29 @@ jobs:
           END
           )
 
-          result=$(echo "$newjson" | jq '. | @json' )
+          # unsupported restrictions outside of Github organisations
+          if [[ $is_org =~ false ]]; then
+              newjson="$(echo "${newjson}" | jq -r 'del(.required_pull_request_reviews.dismissal_restrictions, .required_pull_request_reviews.bypass_pull_request_allowances)')"
+          fi
+
+          result=$(echo "${newjson}" | jq '. | @json' )
           echo "result=${result}" >> $GITHUB_OUTPUT
+      # https://github.com/cli/cli/issues/3528
       - name: Apply branch protection rules
         id: apply_branch_protection_rules
         if: ${{ steps.parse_prepare_protection_rules.conclusion == 'success' }}
         run: |
-          status_code="$(curl -o body.json --silent -w "%{http_code}\n" -X PUT "${{ env.BRANCH_PROTECTION_URL }}" \
-            -H 'Accept: application/vnd.github+json' \
-            -H 'Authorization: Bearer ${{ secrets.FLOWZONE_TOKEN }}' \
-            -d '${{ fromJSON(steps.parse_prepare_protection_rules.outputs.result) }}')"
+          result="$(echo '${{ fromJSON(steps.parse_prepare_protection_rules.outputs.result) }}' \
+            | gh api --method PUT ${{ env.BRANCH_PROTECTION_URI }} --input -)"
 
-          result=$(cat body.json | jq '.' )
-          if [[ "$status_code" -ne 200 ]] ; then
-            echo ::error title={url}::"apply_branch_protection_rules failed with ${status_code} ${result} "
+          message="$(echo "${result}" | jq -r .message)"
+
+          if ! [[ $message =~ null ]]; then
+            echo ::error title={url}::"apply_branch_protection_rules failed with ${message} ${result} "
             exit 1
           fi
 
-          if [[  $(echo "$result" | jq '.required_pull_request_reviews.required_approving_review_count') -ne ${{ inputs.required_approving_review_count }} ]]; then
+          if [[ $(echo "$result" | jq '.required_pull_request_reviews.required_approving_review_count') -ne ${{ inputs.required_approving_review_count }} ]]; then
             echo ::error title={url}::"branch protection required_pull_request_reviews.required_approving_review_count not set to ${{ inputs.required_approving_review_count }}"
```

Comment:
I know you didn't write this error, but is there something more helpful we can print here?

Reply:
@klutchell I am struggling to understand what that entire block does:

```
...
          if [[ $(echo "$result" | jq '.required_pull_request_reviews.required_approving_review_count') -ne ${{ inputs.required_approving_review_count }} ]]; then
            echo ::error title={url}::"branch protection required_pull_request_reviews.required_approving_review_count not set to ${{ inputs.required_approving_review_count }}"
            exit 1
          fi

          DELIMITER=$(echo $RANDOM | md5sum | head -c 32)
          echo "result<<${DELIMITER}" >> $GITHUB_OUTPUT
          echo "${result}" >> $GITHUB_OUTPUT
          echo "${DELIMITER}" >> $GITHUB_OUTPUT
```

As fas as I can see, `DELIMITER` isn't used anywhere. Shall I just remove the whole lot?